core/lib/config/src/configs/snapshot_recovery.rs

```diff
@@ -0,0 +1,41 @@
+use std::num::NonZeroUsize;
+
+use serde::Deserialize;
+use zksync_basic_types::L1BatchNumber;
+
+#[derive(Debug, Clone, PartialEq, Deserialize, Default)]
+pub struct TreeRecoveryConfig {
+    /// Approximate chunk size (measured in the number of entries) to recover in a single iteration.
+    /// Reasonable values are order of 100,000 (meaning an iteration takes several seconds).
+    ///
+    /// **Important.** This value cannot be changed in the middle of tree recovery (i.e., if a node is stopped in the middle
+    /// of recovery and then restarted with a different config).
+    pub chunk_size: Option<u64>,
+    /// Buffer capacity for parallel persistence operations. Should be reasonably small since larger buffer means more RAM usage;
+    /// buffer elements are persisted tree chunks. OTOH, small buffer can lead to persistence parallelization being inefficient.
+    ///
+    /// If not set, parallel persistence will be disabled.
+    pub parallel_persistence_buffer: Option<NonZeroUsize>,
```

Comment:
Got it. It's still an experimental parameter though, and if we e.g. decide to re-parameterize it, Protobuf-based backward compatibility checks can get in the way. (To be fair, just removing the param would work fine, although it would produce some visual noise in the form of a reserved Protobuf field.)

Reply:
What is your suggestion here? 
We can't just mark it as experimental

Reply:
also, i think as well as we do with database, we can cleanup all reserved after the upgrade

Reply:
One option would be to define it in a separate struct (e.g., `ExperimentalSnapshotRecoveryConfig` with a `tree_ parallel_persistence_buffer` field) in the `experimental` module, attach it as a field to `SnapshotRecoveryConfig`, and correspondingly define it in the `zksync.config.experimental` package in Protobuf. This package [is excluded](https://github.com/matter-labs/zksync-era/blob/dd2c941e94b5fde9064d9ff4394a00a4b0707fbf/.github/workflows/protobuf.yaml#L77) from backward compatibility checks.

Reply:
It's a bit painful, but i've done it

Reply:
I feel it's painful because you've decided to make it painful 🙃 IMO, it'd be much simpler if you had the same config structure in Protobuf and here, as I've suggested. Well, it'd be even simpler if we didn't have Protobuf config altogether, but we've been over this.

Reply:
i think, having the same structure will make everything even harder. 
Let's stick to this approach..